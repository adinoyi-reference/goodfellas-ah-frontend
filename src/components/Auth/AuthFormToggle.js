import React, { Component } from 'react';
import { Link } from 'react-router-dom';

// eslint-disable-next-line react/prefer-stateless-function
class AuthFormToggle extends Component {
  render() {
    const { children: { props: { formtype } } } = this.props;
    const activeSignin = (formtype === 'signin' ? 'active' : null);
    const activeSignup = (formtype === 'signup' ? 'active' : null);
    return (
      <div className="auth-buttons">
        <Link className="both-links" to="/auth/signup">
          <button type="button" className={`buttons sign-up ${activeSignup}`}>SIGN UP</button>
        </Link>
        <Link className="both-links" to="/auth/signin">
          <button type="button" className={`buttons sign-in ${activeSignin}`}>SIGN IN</button>
        </Link>
      </div>
    );
  }
}

export default AuthFormToggle;
