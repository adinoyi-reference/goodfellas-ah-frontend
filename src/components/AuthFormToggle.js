import React, { Component } from 'react';
import { Link } from 'react-router-dom';

// eslint-disable-next-line react/prefer-stateless-function
class AuthFormToggle extends Component {
  render() {
    const { children: { props: { formtype } } } = this.props;
    const activeSignin = (formtype === 'signin' ? 'active' : null);
    const activeSignup = (formtype === 'signup' ? 'active' : null);
    return (
      <div className="signup-buttons">
        <Link className="both-links" to="/auth/signup">
          <input type="button" className={`buttons sign-up ${activeSignup}`} value="SIGN UP" />
        </Link>
        <Link className="both-links" to="/auth/signin">
          <input type="button" className={`buttons sign-in ${activeSignin}`} value="SIGN IN" />
        </Link>
      </div>
    );
  }
}

export default AuthFormToggle;
