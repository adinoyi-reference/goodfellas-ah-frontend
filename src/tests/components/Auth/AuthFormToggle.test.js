import React from 'react';
import { shallow } from 'enzyme';
import AuthFormToggle from '../../../components/auth/AuthFormToggle';

const setup = () => {
  const props = {
    children: { props: { formtype: 'signup' } },
  };

  const enzymeWrapper = shallow(<AuthFormToggle {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};

describe('components', () => {
  describe('ProfileToolbar component', () => {
    it('should render', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
