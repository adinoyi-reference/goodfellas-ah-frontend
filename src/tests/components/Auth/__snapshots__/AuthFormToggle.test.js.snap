// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ProfileToolbar component should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthFormToggle>
    Object {
      "props": Object {
        "formtype": "signup",
      },
    }
  </AuthFormToggle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="both-links"
          replace={false}
          to="/auth/signup"
        >
          <button
            className="buttons sign-up active"
            type="button"
          >
            SIGN UP
          </button>
        </Link>,
        <Link
          className="both-links"
          replace={false}
          to="/auth/signin"
        >
          <button
            className="buttons sign-in null"
            type="button"
          >
            SIGN IN
          </button>
        </Link>,
      ],
      "className": "auth-buttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="buttons sign-up active"
            type="button"
          >
            SIGN UP
          </button>,
          "className": "both-links",
          "replace": false,
          "to": "/auth/signup",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SIGN UP",
            "className": "buttons sign-up active",
            "type": "button",
          },
          "ref": null,
          "rendered": "SIGN UP",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="buttons sign-in null"
            type="button"
          >
            SIGN IN
          </button>,
          "className": "both-links",
          "replace": false,
          "to": "/auth/signin",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SIGN IN",
            "className": "buttons sign-in null",
            "type": "button",
          },
          "ref": null,
          "rendered": "SIGN IN",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="both-links"
            replace={false}
            to="/auth/signup"
          >
            <button
              className="buttons sign-up active"
              type="button"
            >
              SIGN UP
            </button>
          </Link>,
          <Link
            className="both-links"
            replace={false}
            to="/auth/signin"
          >
            <button
              className="buttons sign-in null"
              type="button"
            >
              SIGN IN
            </button>
          </Link>,
        ],
        "className": "auth-buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="buttons sign-up active"
              type="button"
            >
              SIGN UP
            </button>,
            "className": "both-links",
            "replace": false,
            "to": "/auth/signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SIGN UP",
              "className": "buttons sign-up active",
              "type": "button",
            },
            "ref": null,
            "rendered": "SIGN UP",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="buttons sign-in null"
              type="button"
            >
              SIGN IN
            </button>,
            "className": "both-links",
            "replace": false,
            "to": "/auth/signin",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SIGN IN",
              "className": "buttons sign-in null",
              "type": "button",
            },
            "ref": null,
            "rendered": "SIGN IN",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
